pkg/provider/bitwarden/client.go

```diff
@@ -231,6 +236,33 @@ func (p *Provider) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecre
 	return secretData, nil
 }
 
+func (p *Provider) parseYamlSecretData(data []byte) (map[string][]byte, error) {
+	kv := make(map[string]any)
+	if err := yaml.Unmarshal(data, &kv); err != nil {
+		return nil, fmt.Errorf("error unmarshalling secret: %w", err)
+	}
+
+	secretData := make(map[string][]byte)
+	for k, v := range kv {
```

Comment:
Looking at this, there might be some more cases that we should consider:

https://github.com/external-secrets/external-secrets/blob/abd7b776118dcd8006e170c7676b6741c7e8b114/pkg/utils/utils.go#L309-L334

Though i think this is not usable since it reverts back to json.Marshal, instead of yaml.Marshal.

Reply:
Dude, I'm not sure I understand what you are suggesting here. :D Are you suggesting to make use of the utils thing instead or that we are missing things here, or that we should potentially allow all providers to have this functionality? :D

Reply:
Sorry, lack of time == lack of clarity :disappointed: 

This implementation supports `string`, `[]byte` and `map[string]any`, what about goofy types like `[]string`, `int`, `float64` (or others?). Above mentioned implementation supports a few more types.

Not saying to support `yaml` everywhere (looked at that at some point in time but without success)

Reply:
Ahhhhhhhhhha. Now I see. Yeah, you're right. Very good point. Thanks.

Reply:
@moolen So I have the default case for that saying `secretData[k] = []byte(fmt.Sprintf("%v", t)) // Convert to string and then []byte`.

Which essentially will result in whatever the other types are in a byte slice.

So `key: true` and than fetching key will result in `true` in bytes. Or `key: 1.23` will just result in the byte slice `1.23`.

What else would you need?

Reply:
@moolen Tentative Ping. :)

Reply:
I see and tested it, looks good to me!